### DISCUSSION TITLE
Global interceptor to do things like only send error/critical logs but ignore other types of logs

### DISCUSSION THREAD

**Discussion by vesper8** [spatie/ray#645] [General]
I'm wondering if there's currently a way to split what kind of logs is sent to ray. Currently it either sends all logs or none at all, but I would like to only send logs based on their level.

I would also like a means of intercepting ray logs (in Laravel) before they are sent, so that I could be free to do things like `only send this to ray if this or if that`.

I know the new `->when()` method can be helpful with that, but I want to be able to use `->when()` for things like errors or logs where I am not triggering ray myself but it's being triggered by a system event.

A global interceptor of sorts would do the trick here. Is there anything like that at the moment?

Thank you

**Comment by freekmurze** [comment]
There isn't anything right now atm, but I'm if there is a simple way to add this, I'm open to a PR.
